app/components/amplience/product-item/index.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage} from 'react-intl'
+
+// Chakra Components
+import {Box, Fade, Flex, Stack, Text} from '@chakra-ui/react'
+
+// Project Components
+import {HideOnDesktop, HideOnMobile} from '../../responsive'
+import ItemVariantProvider from '../../item-variant'
+import CartItemVariantImage from '../../item-variant/item-image'
+import CartItemVariantName from '../../item-variant/item-name'
+import CartItemVariantAttributes from '../../item-variant/item-attributes'
+import CartItemVariantPrice from '../../item-variant/item-price'
+import LoadingSpinner from '../../loading-spinner'
+import QuantityPicker from '../../quantity-picker'
+
+// Utilities
+import {noop} from '../../../utils/utils'
+
+// Hooks
+import {useProduct} from '../../../hooks'
+
+/**
+ * Component representing a product item usually in a list with details about the product - name, variant, pricing, etc.
+ * @param {Object} product Product to be represented in the list item.
+ * @param {node} primaryAction Child component representing the most prominent action to be performed by the user.
+ * @param {node} secondaryActions Child component representing the other actions relevant to the product to be performed by the user.
+ * @param {func} onItemQuantityChange callback function to be invoked whenever item quantity changes.
+ * @param {boolean} showLoading Renders a loading spinner with overlay if set to true.
+ * @returns A JSX element representing product item in a list (eg: wishlist, cart, etc).
+ */
+const ProductItem = ({
+    product,
+    primaryAction,
+    secondaryActions,
+    onItemQuantityChange = noop,
+    showLoading = false
+}) => {
+    const {
+        stepQuantity,
+        showInventoryMessage,
+        inventoryMessage,
+        quantity,
+        setQuantity
+    } = useProduct(product)
+
+    return (
+        <Box position="relative" data-testid={`sf-cart-item-${product.productId}`}>
+            <ItemVariantProvider variant={product}>
+                {showLoading && <LoadingSpinner />}
+                <Stack layerStyle="cardBordered" align="flex-start">
+                    <Flex width="full" alignItems="flex-start" backgroundColor="white">
+                        <CartItemVariantImage width={['88px', '136px']} mr={4} />
+                        <Stack spacing={3} flex={1}>
+                            <Stack spacing={1}>
+                                <CartItemVariantName />
+                                <CartItemVariantAttributes />
+                                <HideOnDesktop>
+                                    <Box marginTop={2}>
+                                        <CartItemVariantPrice align="left" />
+                                    </Box>
+                                </HideOnDesktop>
+                            </Stack>
+
+                            <Flex align="flex-end" justify="space-between">
+                                {/* <Stack spacing={1}>
```

Comment:
Lots of commented code here. Should this be removed?

Reply:
removed